Magnetometer fix noise initialization

GorgiAstro opened the pull request:
* **Tickets addressed:** #411
* **Review:** By commit
* **Merge strategy:** Merge (no squash) 

## Description
- Fix #411 by moving the line `this->noiseModel.setRNGSeed(this->RNGSeed); ` from the magnetometer's constructor to the `Reset` method
- Also, fix the beginning of `Magnetometer::applySensorErrors` which checks that the 3 components of the noise std are initialized. Previously, unlike what is written in the comment at line 126, the noise model was disabled if all three components of this->senNoiseStd are uninitialized. Now the code already disables the noise model if only one of the 3 components of this->senNoiseStd is uninitialized.

## Verification
The noise of two magnetometers with different RNG seeds were plotted: before the fix, they were correlated, and after the fix they are not correlated anymore.

The magnetometer's unit test was not edited to check that the noise model was set up with the proper RNG seed because the `noiseModel` member variable is private within the `Magnetometer` class.

schaubh reviewed (changes requested):
Please add release notes to 
`docs/source/Support/bskReleaseNotes.rst` 
for the latest beta cycle.  

Otherwise I'm good with these suggested changes.

GorgiAstro commented:
Thanks, I added an explanation of the changes to the release notes.

patkenneally commented:
If you can manage this branch should be rebased. You third commit is a merge of develop into your branch something we work hard to avoid being in the git history. This is outlined in the [CONTRIBUTING.md](https://github.com/AVSLab/basilisk/blob/develop/CONTRIBUTING.md). Happy to assist if needed.

patkenneally commented:
This branch should be rebased. Your third commit is a merge of develop branch into your branch, something we work hard to avoid being in the git history. It should be removed. This is outlined in the [CONTRIBUTING.md](https://github.com/AVSLab/basilisk/blob/develop/CONTRIBUTING.md). Happy to assist if needed.

GorgiAstro commented:
> This branch should be rebased. Your third commit is a merge of develop branch into your branch, something we work hard to avoid being in the git history. It should be removed. This is outlined in the [CONTRIBUTING.md](https://github.com/AVSLab/basilisk/blob/develop/CONTRIBUTING.md). Happy to assist if needed.

Sorry about that, I saw this commit, it was not from me, it looks like it was triggered by Prof. Schaub from GitHub, so I thought it is usual practice on this repo. Anyways, I now rebased on the latest `develop` and updated via `force-push` :)

schaubh commented:
I hit the merge button last week by accident.  It was labeled differently where I thought it would do a rebase, but clearly it didn't.  Thanks for doing the rebase and force push.  I triggered the CI checks to run.  Should be all good now.  They take about 1h to complete.

GorgiAstro commented:
> I hit the merge button last week by accident. It was labeled differently where I thought it would do a rebase, but clearly it didn't. Thanks for doing the rebase and force push. I triggered the CI checks to run. Should be all good now. They take about 1h to complete.

That explains it then. Thanks for the swift response to my merge request!
Cheers
Clément
